tests/unit/test_adapter.py

```diff
@@ -44,39 +45,192 @@ def _get_target_databricks_sql_connector(self, project):
             },
         )
 
-    def test_databricks_sql_connector_connection(self):
+    def _get_target_databricks_sql_connector_catalog(self, project):
+        return config_from_parts_or_dicts(
+            project,
+            {
+                "outputs": {
+                    "test": {
+                        "type": "databricks",
+                        "schema": "analytics",
+                        "catalog": "main",
+                        "host": "yourorg.databricks.com",
+                        "http_path": "sql/protocolv1/o/1234567890123456/1234-567890-test123",
+                        "token": "dapiXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+                        "session_properties": {"spark.sql.ansi.enabled": "true"},
+                    }
+                },
+                "target": "test",
+            },
+        )
+
+    def test_two_catalog_settings(self):
+        with self.assertRaisesRegexp(
+            dbt.exceptions.DbtProfileError,
+            "Got duplicate keys: \\(`databricks.catalog` in session_properties\\)"
+            ' all map to "database"',
+        ):
+            config_from_parts_or_dicts(
+                self.project_cfg,
+                {
+                    "outputs": {
+                        "test": {
+                            "type": "databricks",
+                            "schema": "analytics",
+                            "catalog": "main",
+                            "host": "yourorg.databricks.com",
+                            "http_path": "sql/protocolv1/o/1234567890123456/1234-567890-test123",
+                            "token": "dapiXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+                            "session_properties": {
+                                CATALOG_KEY_IN_SESSION_PROPERTIES: "catalog",
+                                "spark.sql.ansi.enabled": "true",
+                            },
+                        }
+                    },
+                    "target": "test",
+                },
+            )
+
+    def test_database_and_catalog_settings(self):
+        with self.assertRaisesRegexp(
+            dbt.exceptions.DbtProfileError,
+            'Got duplicate keys: \\(catalog\\) all map to "database"',
```

Comment:
How about "Cannot specify both `catalog` and `database`"?

Reply:
That's from `dbt-spark` or `dbt-core`. I think we should follow the error message.

Reply:
Sounds good!